app/models/form_profile.rb

```diff
@@ -77,34 +97,59 @@ def initialize_identity_information(user)
         suffix: user.va_profile.suffix
       },
       date_of_birth: user.birth_date,
-      gender: user.gender
+      gender: user.gender,
+      ssn: user.ssn
     )
   end
 
   def initialize_contact_information(user)
+    address = {
+      street: user.va_profile.address.street,
+      street2: nil,
+      city: user.va_profile.address.city,
+      state: user.va_profile.address.state,
+      postal_code: user.va_profile.address.postal_code,
+      country: user.va_profile.address.country
+    } if user.va_profile.address
     FormContactInformation.new(
-      address: {
-        street: user.va_profile.address.street,
-        street2: nil,
-        city: user.va_profile.address.city,
-        state: user.va_profile.address.state,
-        postal_code: user.va_profile.address.postal_code,
-        country: user.va_profile.address.country
-      },
+      address: address,
       home_phone: user.va_profile.home_phone
     )
   end
 
   def generate_prefill(mappings)
-    mappings.map do |k, v|
+    result = mappings.map do |k, v|
       method_chain = v.map(&:to_sym)
       { k.camelize(:lower) => call_methods(method_chain) }
     end.reduce({}, :merge)
+    clean!(result)
   end
 
   def call_methods(methods)
-    methods.inject(self) { |a, e| a.send e }
+    methods.inject(self) { |a, e| a.send e }.as_json
   rescue NoMethodError
     nil
   end
+
+  def clean!(value)
```

Comment:
think this could be a bit cleaner as a case, and do you need the bool numeric condition?
``` ruby
def clean!(value)
  case value
  when Hash
    clean_hash!(value)
  when Array
    result = value.map do |item|
      clean!(item)
    end
    result.delete_if(&:blank?)
  else
    value.to_s
  end
end
```
^ btw that and original seem to randomly fail specs; `--seed 6551` passes `--seed 14807` fails. If cleaning useful beyond this solution it could go in common as a class of its own or on Hash.